lib/foreman_hits/async/upload.rb

```diff
@@ -0,0 +1,53 @@
+module ForemanHits
+  module Async
+    class Upload < ::Actions::EntryAction
+      def plan(host, uuid, payload = {})
+        plan_self(host_id: host.id, uuid: uuid, payload: payload)
+      end
+
+      def run
+        host = Host.find(input[:host_id])
+        payload = input[:payload]
+        update_facets(host, input[:uuid])
+        update_hits(host, payload)
+        update_rules_and_resolutions(payload)
+        update_details(host, payload)
```

Comment:
Database actions shouldn't be in `run` since `run` doesn't run in a transaction, right? Can it maybe be split into sub-actions and moved to `def plan` / `def finalize` ?

Reply:
Moving it to a service file instead of a dynflow task